Challenge-4/Bot_1/src/helper.ts

```diff
@@ -49,12 +43,20 @@ export default class Helper {
           blockTag: blockNumber,
         });
 
-        const collateralName =
-          "Collateral Asset - " +
-          (await this.getCollateralName(
-            collateralAssetInfo.asset,
-            blockNumber,
-          ));
+        let collateralName =
+          this.collateralNamesCacheObject[collateralAssetInfo.asset];
+
+        if (!collateralName) {
+          collateralName =
+            "Collateral Asset - " +
+            (await this.getCollateralName(
+              collateralAssetInfo.asset,
+              blockNumber,
+            ));
+          this.collateralNamesCacheObject[collateralAssetInfo.asset] =
+            collateralName;
+        }
```

Comment:
Not necessary but something for you to think about and that's to have this logic in the `getCollateralName()` function.